detectable/src/main/java/com/synopsys/integration/detectable/detectables/rubygems/gemlock/parse/GemlockParser.java

```diff
@@ -68,21 +69,21 @@
 
     private GemfileLockSection currentSection = NONE;
 
-    private List<String> encounteredDependencies = new ArrayList<>();
-    private List<String> resolvedDependencies = new ArrayList<>();
+    private Set<String> encounteredDependencies = new HashSet<>();
```

Comment:
We had a list that we would iterate through all the names.  I thought using hashsets for the strings would be a faster implementation for the contains method call that's used.